wallet/src/events.rs

```diff
@@ -8,11 +8,14 @@ use serde::de::{SeqAccess, Visitor};
 use serde::{Deserialize, Deserializer};
 use std::cell::RefCell;
 use std::cmp::min;
+use std::collections::VecDeque;
 use std::fmt::{self, Formatter};
+use std::ops::Range;
 
 #[derive(CandidType, Clone, Default, Deserialize)]
 pub struct EventBuffer {
-    pub events: Vec<Event>,
+    pub events: VecDeque<Event>,
+    pub culled: Option<usize>,
```

Comment:
1. What's the rationale for storing this here, as opposed to calculating it (as `total` does)?
2. Does anything special need to happen during upgrade? Unless I'm missing something, I only see it being filled with Some(0), which will be correct for this upgrade, but not for the next one.

Reply:
You're right, I had it in my head before I started implementing it. And if I *were* to keep it, which I'm not, it'll be filled with Some(0), and then the next recorded event would have set it correctly as however many events were culled.